packages/wrangler/src/queues/cli/commands/update.ts

```diff
@@ -0,0 +1,110 @@
+import { readConfig } from "../../../config";
+import { CommandLineArgsError } from "../../../index";
+import { logger } from "../../../logger";
+import { getQueue, updateQueue } from "../../client";
+import {
+	MAX_DELIVERY_DELAY_SECS,
+	MAX_MESSAGE_RETENTION_PERIOD_SECS,
+	MIN_DELIVERY_DELAY_SECS,
+	MIN_MESSAGE_RETENTION_PERIOD_SECS,
+} from "../../constants";
+import { handleFetchError } from "../../utils";
+import type {
+	CommonYargsArgv,
+	StrictYargsOptionsToInterface,
+} from "../../../yargs-types";
+import type { PostQueueBody, QueueSettings } from "../../client";
+
+export function options(yargs: CommonYargsArgv) {
+	return yargs
+		.positional("name", {
+			type: "string",
+			demandOption: true,
+			description: "The name of the queue",
+		})
+		.options({
+			"delivery-delay-secs": {
+				type: "number",
+				describe:
+					"How long a published message should be delayed for, in seconds. Must be between 0 and 42300",
+			},
+			"message-retention-period-secs": {
+				type: "number",
+				describe:
+					"How long to retain a message in the queue, in seconds. Must be between 60 and 1209600",
+			},
+		});
+}
+
+function updateBody(
+	args: StrictYargsOptionsToInterface<typeof options>,
+	currentSettings?: QueueSettings
+): PostQueueBody {
+	const body: PostQueueBody = {
+		queue_name: args.name,
+	};
+
+	if (Array.isArray(args.deliveryDelaySecs)) {
+		throw new CommandLineArgsError(
+			"Cannot specify --delivery-delay-secs multiple times"
+		);
+	}
+
+	if (Array.isArray(args.messageRetentionPeriodSecs)) {
+		throw new CommandLineArgsError(
+			"Cannot specify --message-retention-period-secs multiple times"
+		);
+	}
+
+	body.settings = {};
+
+	if (args.deliveryDelaySecs != undefined) {
+		if (
+			args.deliveryDelaySecs < MIN_DELIVERY_DELAY_SECS ||
+			args.deliveryDelaySecs > MAX_DELIVERY_DELAY_SECS
+		) {
+			throw new CommandLineArgsError(
```

Comment:
sanity check: from 👇  comment [here](https://github.com/cloudflare/workers-sdk/blob/main/packages/wrangler/src/index.ts#L199-L201)

```
// We handle errors ourselves in a try-catch around `yargs.parse`.
// If you want the "help info" to be displayed then throw an instance of `CommandLineArgsError`.
// Otherwise we just log the error that was thrown without any "help info".
```
just making sure that this was the intention here when throwing `CommandLineArgsError`